lib/std/meta.zig

```diff
@@ -103,6 +103,106 @@ test "std.meta.stringToEnum" {
     testing.expect(null == stringToEnum(E1, "C"));
 }
 
+pub fn errorInSet(err: anytype, comptime Set: type) bool {
+    const set_errs = comptime blk: {
+        const err_infos = @typeInfo(Set).ErrorSet orelse
+            return true; // global error set!
+        var errs: [err_infos.len]Set = undefined;
+        inline for (err_infos) |err_info, i| {
+            errs[i] = @field(Set, err_info.name);
+        }
+
+        break :blk errs;
+    };
+
+    for (set_errs) |set_err| {
+        if (err == @errSetCast(anyerror, set_err)) {
+            return true;
+        }
+    }
+
+    return false;
+}
+
+test "std.meta.errorInSet" {
+    testing.expect(!errorInSet(error.A, error{}));
+    testing.expect(!errorInSet(error.B, error{A}));
+    testing.expect(errorInSet(error.A, error{A}));
+    testing.expect(!errorInSet(error.D, error{ A, B, C }));
+    testing.expect(errorInSet(error.B, error{ A, B, C }));
+    testing.expect(errorInSet(error.B, anyerror));
+    testing.expect(errorInSet(error.ErrorName, anyerror));
+    testing.expect(errorInSet(error.OtherError, anyerror));
+}
+
+fn lookupDeclType(comptime T: type, comptime names: []const []const u8) ?type {
+    comptime var Running = T;
+    inline for (names) |name, i| {
+        if (@hasDecl(Running, name)) {
+            const next = @field(Running, name);
+            if (@TypeOf(next) == type) {
+                Running = next;
+                continue;
+            }
+            return if (i == names.len - 1)
+                @TypeOf(next)
+            else
+                @compileError("lookupDecl expected type and found value!");
+        } else return null;
+    }
+
+    return type;
+}
+
+/// Get a decl on T by indexing into each field. Returns null if one of the
+/// sub-decls doesn't exist, `compileError` if a sub-decl other than the last one
+/// is a value instead of a type.
+pub fn lookupDecl(comptime T: type, comptime names: []const []const u8) ?(lookupDeclType(T, names) orelse type) {
```

Comment:
I also added this utility function. This is generally useful for the `if (@hasDecl(root, "name")) root.name else default` pattern. This ended up being considerably grosser to implement than I expected and I am considering removing it.